host.go

```diff
@@ -499,3 +522,38 @@ func (h *Host) SaveConfig() error {
 	}
 	return nil
 }
+
+func (h *Host) SetInstanceHostnameUbuntu() error {
+	log.Debugf("Setting hostname for: %s", h.Name)
+	cmd, err := h.Driver.GetSSHCommand(fmt.Sprintf(
```

Comment:
We could merge the two host name functions and supply a path to the hostname file.